src/export/StructureDefinitionExporter.ts

```diff
@@ -168,48 +168,75 @@ export class StructureDefinitionExporter implements Fishable {
   }
 
   /**
-   * Does any necessary preprocessing of extensions.
+   * Does any necessary preprocessing of profiles and extensions.
    * @param {Extension} fshDefinition - The extension to do preprocessing on. It is updated directly based on processing.
    */
-  private preprocessExtension(fshDefinition: Extension): void {
-    let hasRuleAppliedToValueX = false;
-    let hasRuleAppliedToExtension = false;
+  private preprocessStructureDefinition(
+    fshDefinition: Extension | Profile,
+    isExtension: boolean
+  ): void {
+    const inferredCardRulesMap = new Map(); // key is the rule, value is a boolean of whether it should be set
     fshDefinition.rules.forEach(rule => {
-      if (rule.path.startsWith('extension')) {
-        if (hasRuleAppliedToValueX) {
-          logger.error(
-            `Extension ${fshDefinition.name} cannot have both a value and sub-extensions`,
-            rule.sourceInfo
+      const rulePathParts = splitOnPathPeriods(rule.path);
+      rulePathParts.forEach((pathPart, i) => {
+        const initialPath = rulePathParts.slice(0, i).join('.');
+        const basePath = `${initialPath}${initialPath ? '.' : ''}`;
+        const previousPathPart = rulePathParts.slice(i - 1, i)[0];
```

Comment:
Isn't this just a really complicated way of saying `rulePathParts[i-1]`?

Reply:
Hm well yes. I think this shows the hole I truly dug myself into on this. I'll change it.